cityofaustin/atd-vz-data
# Cascade delete crash notes when parent record is deleted

## Associated issues
Closes cityofaustin/atd-data-tech/issues/11030

Upon deleting a temp crash and creating a new temp crash that reused the same crash_id (temp crash_id's are created by incrementing by 1 from the highest crash_id), the notes from the first temp crash were carrying forward into the new crash because the crash_note for that crash_id was not deleted in the DB.

There are no code changes for this PR but I wanted to open it so y'all could test the fix I made for the issue above and also so I can outline the changes I made to the DB. @frankhereford and I had a discussion about whether I needed to make the usual SQL files for these minor database updates and want to have a discussion with the rest of the dev team about the necessity of these files/whether we should continue this practice considering we haven't been consistent with them and haven't used them much (to our knowledge?)

Changes:
- Added the following foreign key - crash_id → atd_txdot_crashes . crash_id - crash_notes_crash_id_fkey with:
    - Cascade on delete violation
    - Cascade on update violation
- Added an object relationship: crash_notes . crash_id  → atd_txdot_crashes . crash_id
- Added an array relationship: crash_notes . crash_id  → atd_txdot_crashes . crash_id

## Testing
**URL to test:** 
You can test this in VZE staging (https://visionzero-staging.austinmobility.io/editor/#/tools/create_crash_record)

**Steps to test:**

1. Navigate to tools -> create crash record
2. Create a new crash record, make note of the crash_id that was generated
3. Add a new note to that temp crash record
4. Go back to tools -> create crash record, scroll down and delete the temp crash you created
5. Wait a minute to give the database time to make the deletes
6. Create another crash record, it should have the same crash_id but the note you made earlier shouldn't be present in the crash details page for the new crash.
7. You could also be checking the hasura staging console to double-check that the crash in the atd_txdot_crashes table and the note in the crash_notes table is deleted.

Once this PR is approved I will make the same changes in the hasura prod console. Also, to fix Xavier's specific issue with crash 1026, I can delete that crash note in the DB.

---
#### Ship list
